### ISSUE TITLE
Filters in frontend disappear when you add a sequence to Smile_ElasticsuiteCatalog in a module

### ISSUE THREAD

**Issue by Tokipudi** [Smile-SA/elasticsuite#886]
### Preconditions
Magento Version : _EE 2.2.2_
ElasticSuite Version : _2.5.8_
Environment : _Developer_

### Steps to reproduce
1. Create a new module
2. Add a sequence to **Smile_ElasticsuiteCatalog**

### Expected result
The new module works properly

### Actual result
The filters in the frontend, on the category page, disappear. 

### What is causing the issue
It is caused by the fact that **_Smile_ElasticsuiteCatalog_** gets loaded before **_Magento_LayeredNavigationStaging_**. This is due to the fact that **_Smile_ElasticsuiteCatalog_** has a sequence to **_Magento_LayeredNavigation_**, instead of  **_Magento_LayeredNavigationStaging_**.

**_Smile_ElasticsuiteCatalog_** has a virtualType that requires **_Magento_LayeredNavigationStaging_** to be loaded first for it to work :
```
<virtualType name="Magento\LayeredNavigationStaging\Block\Navigation\Search" type="Smile\ElasticsuiteCatalog\Block\Navigation">
    <arguments>
        <argument name="filterList" xsi:type="object">searchFilterList</argument>
    </arguments>
</virtualType>
```
You can find it in `frontend/etc/di.xml`

### How to solve the issue
Simply change the line `<module name="Magento_LayeredNavigation" />` in `Smile\ElasticsuiteCatalog\etc\module.xml` by `<module name="Magento_LayeredNavigationStaging" />`, so that **_Magento_LayeredNavigationStaging_**, which also loads **_Magento_LayeredNavigation_** is always loaded before **_Smile_ElasticsuiteCatalog_**.

**Comment by Tokipudi** [comment]
Thank you for your answer. We decided to simply add the sequence needed on every module that needs a sequence towards **Smile_ElasticsuiteCatalog**

I was also wondering something : is there any example on how to use function scores ? I can't seem to find any in the wiki, nor in the code, even though it seems like you've implemented some classes that handle it.

Basically, I would need to be able to use function scores for something similar to [this](https://www.elastic.co/guide/en/elasticsearch/guide/current/boosting-by-popularity.html)

Thank you for your time

**Comment by afoucret** [comment]
Hi @Tokipudi,

I close since you have found a workaround and there is no way for us to prevent this to append.
A good option would be to have optional dependencies in the Magento sequence module.

I will check if it is possible to have propose something to Magento ...

**Comment by afoucret** [comment]
Reopen since it seems that we can add a non existing module into the module sequence definition without breaking anything. Dependency check is done using composer.

We should be able to add the Magento_LayeredNavigationStaging into the Smile_ElasticsuiteCatalog sequence without breaking community edition.

Note: it is an hack. Maybe Magento will change this behavior one day.

**Comment by afoucret** [comment]
PR have been merged and will be part of the 2.5.9 upcoming maintenance release.
